core/src/impl/Kokkos_Memory_Fence.hpp

```diff
@@ -54,6 +54,8 @@ void memory_fence() {
   __threadfence();
 #elif defined(KOKKOS_ENABLE_ROCM_ATOMICS)
   amp_barrier(CLK_LOCAL_MEM_FENCE | CLK_GLOBAL_MEM_FENCE);
+#elif defined(KOKKOS_ENABLE_HIP_ATOMICS)
+  __threadfence();
```

Comment:
You didn't want the `memory_fence()` changes in a separate PR?

Reply:
I greped where there was a difference between rocm macro and hip macro and that was one of them. I can split the commit.

Reply:
Your choice.  Merge you are ready.